pages/installation.md

```diff
@@ -16,17 +16,304 @@ permalink: /installation/
+RABBITMQ_URL=amqp://oop:somepassword@localhost
+OOP_ERROR_EXCHANGE_NAME=oop.errors
+OOP_JSON_ERROR_Q=oop.errors.json
+OOP_EXCHANGE_NAME=oop
+OOP_QUEUE_PREFETCH_LIMIT=5
+
+# Gateway
+OOP_GATEWAY_OUTPUT_Q=oop.noauth.raw_messages
+OOP_LISTEN_PORT=3000
+OOP_DB_ADDRESS=http://admin:admin@localhost:5984
+
+# Authenticator
+OOP_AUTHENTICATOR_INPUT_Q=oop.noauth.raw_messages
+OOP_AUTHENTICATOR_OUTPUT_Q=oop.hasauth.messages
+OOP_CORE_TOKEN=foobar
+OOP_CORE_DEVICE_UPDATE_EXCHANGE=oop.core.devices
+
+# Tempr
+OOP_TEMPR_INPUT_Q=oop.hasauth.messages
+OOP_TEMPR_OUTPUT_Q=oop.hasauth.temprs
+OOP_CORE_API_URL=http://{CORE_IP_ADDRESS}:9001/services/v1
+
+# Scheduler
+OOP_SCHEDULER_OUTPUT_Q=oop.hasauth.messages
+
+# Renderer
+OOP_RENDERER_INPUT_Q=oop.hasauth.temprs
+OOP_ENDPOINTS_EXCHANGE_NAME=oop.endpoints
+OOP_ENDPOINT_Q=oop.endpoints
+
+# Endpoints HTTP
+OOP_ENDPOINTS_HTTP_OUTPUT_Q=oop.hasauth.relay
+OOP_REQUEST_TIMEOUT=1000
+OOP_ENDPOINTS_HTTP_MAX_RETRIES=3
+
+# Relay
+OOP_RELAY_INPUT_Q=oop.hasauth.relay
+OOP_RECURSIVE_TEMPR_Q=oop.hasauth.temprs
+OOP_CORE_RESPONSE_Q=oop.core.transmissions
+
+# Core
+OOP_AMQP_ADDRESS=amqp://localhost
+OOP_RENDERER_PATH=/path/to/renderer/
+
+OOP_CORE_SCHEME=http://
+OOP_CORE_PORT=9001
+OOP_CORE_PATH=/
+
+PORT=9001
+
+OOP_CORE_INTERFACE_SCHEME=http://
+OOP_CORE_INTERFACE_PORT=80
+OOP_CORE_INTERFACE_PATH=/
+OOP_CORE_FROM_ADDRESS=example@mail.co.uk
+
+# Mail setup
+OOP_CORE_FROM_ADDRESS=
+OOP_CORE_SMTP_ADDRESS=
+OOP_CORE_SMTP_PORT=25
+OOP_CORE_SMTP_DOMAIN=
+OOP_CORE_SMTP_USER_NAME=
+OOP_CORE_SMTP_PASSWORD=
+OOP_CORE_SMTP_AUTHENTICATION=
+OOP_CORE_SMTP_ENABLE_STARTTLS_AUTO=
+
+DATABASE_HOST=localhost
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=password
+POSTGRES_DB=oop_core_development
+
+SECRET_KEY_BASE={SECRETKEY}
+```
+
+### oop-core
+
+Clone the application `git clone http://github.com/open-interop/oop-core.git`
+In the app directory install all dependencies using  `bundle install`
+Create a symlink between
+Ensure all variables are set up correctly in the config directory:
+
+###### `database.yml`
+
+```
+# config/database.yml
+
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  host: <%= ENV.fetch('DATABASE_HOST') %>
+  username: <%= ENV.fetch('POSTGRES_USER') %>
+  password: <%= ENV.fetch('POSTGRES_PASSWORD') %>
+  database: <%= ENV.fetch('POSTGRES_DB') %>
+  variables:
+    statement_timeout: 5000
+
+production:
+  <<: *default
+```
+
+###### `storage.yml`
+
+```
+# config/storage.yml
+
+local:
+  service: Disk
+  root: <%= Rails.root.join("storage") %>
+```
+
+###### `secrets.yml`
+
+```
+# config/secrets.yml
+
+production:
+  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
+```
+
+
+Create the database - `bundle exec rails db:create`
+Run the database migrations - `bundle exec rails db:migrate`
+
+
+To allow login from the interface run the following commands from within the `oop-core` directory to create a dummy user:
```

Comment:
Would be good to refer to the script that I made to do this.

Reply:
Whereabouts can I find this?

Reply:
If you run `bin/rails open_interop:setup_initial_account` it runs you through some options.